src/vs/platform/environment/electron-main/environmentMainService.ts

```diff
@@ -64,4 +70,39 @@ export class EnvironmentMainService extends NativeEnvironmentService implements
 
 	@memoize
 	get useCodeCache(): boolean { return !!this.codeCachePath; }
+
+	unsetSnapExportedVariables() {
+		if (!isLinux) {
+			return;
+		}
+		for (const key in process.env) {
+			if (key.endsWith('_VSCODE_SNAP_ORIG')) {
+				const originalKey = key.slice(0, -17); // Remove the _VSCODE_SNAP_ORIG suffix
+				if (this._snapEnv[originalKey]) {
```

Comment:
This seems suspicious.
Is it checking the `_snapEnv[originalKey]` is empty or not?
If it is not empty then it will skip the the complete loop logic following it.

Reply:
`_snapEnv[originalKey]` will not be defined when this function gets called, this is only present when multiple `unsetSnapExportedVariables()` calls are made without calling to `restoreSnapExportedVariables()`, which does not happen today.